civis/io/_tables.py

```diff
@@ -629,11 +641,133 @@ def csv_to_civis(filename, database, table, api_key=None, client=None,
+    sortkey2 : str, optional
+        The second column in a compound sortkey for the table.
+    delimiter : string, optional
+        The column delimiter. One of ``','``, ``'\\t'`` or ``'|'``.
+    headers : bool, optional
+        Whether or not the first row of the file should be treated as
+        headers. The default, ``None``, attempts to autodetect whether
+        or not the first row contains headers.
+    credential_id : str or int, optional
+        The ID of the database credential.  If ``None``, the default
+        credential will be used.
+    polling_interval : int or float, optional
+        Number of seconds to wait between checks for job completion.
+    archive : bool, optional (deprecated)
+        If ``True``, archive the import job as soon as it completes.
+    hidden : bool, optional
+        If ``True`` (the default), this job will not appear in the Civis UI.
+
+    Returns
+    -------
+    results : :class:`~civis.futures.CivisFuture`
+        A `CivisFuture` object.
+
+    Examples
+    --------
+    >>> with open('input_file.csv', 'w') as _input:
+    ...     _input.write('a,b,c\\n1,2,3')
+    >>> fut = civis.io.csv_to_civis('input_file.csv',
+    ...                             'my-database',
+    ...                             'scratch.my_data')
+    >>> fut.result()
+    """
+    if client is None:
+        client = APIClient(api_key=api_key, resources='all')
+    if archive:
+        warnings.warn("`archive` is deprecated and will be removed in v2.0.0. "
+                      "Use `hidden` instead.", FutureWarning)
+
+    schema, table = table.split(".", 1)
+    db_id = client.get_database_id(database)
+    cred_id = credential_id or client.default_credential
+    delimiter = DELIMITERS.get(delimiter)
+    assert delimiter, "delimiter must be one of {}".format(DELIMITERS.keys())
+
+    destination = dict(remote_host_id=db_id, credential_id=cred_id)
+    import_name = 'CSV import to {}.{}'.format(schema, table)
+    import_job = client.imports.post(import_name, 'AutoImport',
+                                     is_outbound=False,
+                                     destination=destination,
+                                     hidden=hidden)
+
+    options = dict(max_errors=max_errors,
+                   existing_table_rows=existing_table_rows,
+                   distkey=distkey, sortkey1=sortkey1, sortkey2=sortkey2,
+                   column_delimiter=delimiter, first_row_is_header=headers)
+
+    client.imports.post_syncs(
+        import_job.id,
+        source=dict(file=dict(id=file_id)),
+        destination=dict(database_table=dict(schema=schema, table=table)),
+        advanced_options=options)
+
+    run = client.jobs.post_runs(import_job.id)
+    fut = CivisFuture(client.jobs.get_runs,
+                      (import_job.id, run['id']),
+                      polling_interval=polling_interval,
+                      client=client,
+                      poll_on_creation=False)
+
+    if archive:
```

Comment:
We've deprecated the `archive` pattern in favor of `hidden`.

Reply:
+1